lib/db.js

```diff
@@ -261,6 +262,34 @@ module.exports = (
     log.trace({ op: 'DB.sessions', uid: uid })
     const promises = [
       this.pool.get('/account/' + uid + '/sessions')
+        .then(sessionTokens => {
+          if (! MAX_AGE_SESSION_TOKEN_WITHOUT_DEVICE) {
+            return sessionTokens
+          }
+
+          const expiredSessionTokens = []
+
+          // Filter out any expired sessions
+          sessionTokens = sessionTokens.filter(sessionToken => {
+            if (sessionToken.deviceId) {
+              return true
+            }
+
+            if (sessionToken.createdAt > Date.now() - MAX_AGE_SESSION_TOKEN_WITHOUT_DEVICE) {
+              return true
+            }
+
+            expiredSessionTokens.push(Object.assign({}, sessionToken, { id: sessionToken.tokenId }))
+            return false
+          })
+
+          if (expiredSessionTokens.length === 0) {
+            return sessionTokens
+          }
+
+          return this.pruneSessionTokens(uid, expiredSessionTokens)
+            .then(() => sessionTokens)
+        })
```

Comment:
I moved this whole block up so that the expiry can take place before we read from Redis.